clearwater-cluster-manager.root/usr/share/clearwater/clearwater-cluster-manager/scripts/mark_node_failed.py

```diff
@@ -61,7 +62,17 @@ def make_key(site, node_type, datatore, etcd_key):
 key = make_key(site, node_type, datastore, etcd_key)
 logging.info("Using etcd key %s" % (key))
 
-error_syncer = EtcdSynchronizer(NullPlugin(key), dead_node_ip, etcd_ip=local_ip, force_leave=True)
+if datastore == "cassandra":
+  if (run_command("dpkg-query -W -f='${Status}' clearwater-cassandra | grep -iq installed") == 0):
+    sys.path.append("/usr/share/clearwater/clearwater-cluster-manager/plugins")
+    from cassandra_failed_plugin import CassandraFailedPlugin
+
+    error_syncer = EtcdSynchronizer(CassandraFailedPlugin(key, dead_node_ip), dead_node_ip, etcd_ip=local_ip, force_leave=True)
+  else:
+    print "You must run mark_node_failed on a node that has Cassandra installed to remove a node from a Cassandra cluster"
+    os._exit(1)
```

Comment:
You've lost the comment about why this is needed. 

Would it be better to refactor most of this script into a `main()` function, and have the outer script handle calling `os._exit`. That confines the evil to one place.